packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -1167,7 +1227,33 @@ export class Domain extends DomainBase implements IDomain {
       throw new Error(`Unknown Elasticsearch version: ${elasticsearchVersion}`);
     }
 
-    const encryptionAtRestEnabled = props.encryptionAtRest?.enabled ?? (props.encryptionAtRest?.kmsKey != null);
+    const masterUserArn = props.advancedSecurity?.masterUserArn;
+    const masterUserName = props.advancedSecurity?.masterUserName;
+
+    const advancedSecurityEnabled = (masterUserArn ?? masterUserName) != null;
+    const internalUserDatabaseEnabled = masterUserName != null;
+    const masterUserPassword =
```

Comment:
I find this quite complex to read.

Reply:
Yeah, and the formatting doesn't help. Could certainly be cleaned up and made more clear!

Reply:
I'll circle back to this tomorrow morning!

Reply:
@adamelmore are you still going to simplify this a bit?

Reply:
@stephanh Sorry, missed this; your changes looked good to me, would you like to see any further simplification?

Reply:
I think it's easier to read now.